Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc

```diff
@@ -122,21 +122,21 @@ CaloTowerConstituentsMapBuilder::parseTextMap( const std::string& filename, Calo
 //algorithm to assign EE cells to HE towers if no text map is provided
 void CaloTowerConstituentsMapBuilder::assignEEtoHE(const CaloGeometry* geometry, CaloTowerConstituentsMap& theMap, const CaloTowerTopology * cttopo){
   //get EE and HE geometries
-  const CaloSubdetectorGeometry* geomEE ( geometry->getSubdetectorGeometry( DetId::Ecal, EcalEndcap ) );
+  const CaloSubdetectorGeometry* geomEE = (geometry->getSubdetectorGeometry(DetId::Ecal, EcalEndcap));
```

Comment:
@bsunanda - please, justify the changes in this file.

Reply:
We had a discussion with Chris and David before making the changes. For detectors like HGCal (also FastTime) CaloCellGeometry is prepared on the fly for each DetId and that was why shared_ptr was suggested. May be Chris can give the right justification. I had to change back and forth many pointers and in that course some of the cosmetic changes came in. I could not restore to the original setting. I am ready to change if the original was indeed better

Reply:
@bsunanda , does it mean that every time the geometry for a DetId is requested, it is built from scratch ?  
Is there some caching so that if it is asked again the same object is returned ?
Is the caching per-event, or per-IOV of the geometry ?

Reply:
I believe in most of the cases (EB, EE, ES, HB, HE, HF, HO, CT) it simply returns the CellPointer stored in geometry. For HGCal, only wafer positions are stored - so it calculates the cell position and returns. With the method you suggested, I understand the overhead for the older detectors is not very high and we did not have the provision for HGCal anyway.

Reply:
my concerns were
  - if there is no caching, why return a `shared_ptr<>` instead of imply returning by value ?
  - if there is per-event caching, what will take care of the clean up and invalidate the cache ? and won't this be a problem for concurrency ?
  - if there is longer-term caching (i.e. per run), won't we eventually end up caching most of the geometry anyway ?